src/main/frontend/components/conversion.cljs

```diff
@@ -0,0 +1,151 @@
+      (let [config-format (state/get-filename-format repo)]
+        (reset! *dir-format config-format)
+        (reset! *solid-format config-format)
+        (reset! *target-format :triple-lowbar)))
+    [:div
+     [:h1.title (t :settings-page/filename-format)]
+     [:div.rounded-md.opacity-70
+      [:p (t :file-rn/filename-desc-1)]
+      [:p (t :file-rn/filename-desc-2)]
+      [:p (t :file-rn/filename-desc-3)]
+      [:p (t :file-rn/filename-desc-4)]]
+     (when (= @*solid-format :legacy)
+       [:div ;; Normal UX stage 1: show the admonition & button for users using legacy format
+        (ui/admonition :warning [:p (t :file-rn/format-deprecated)])
+        (ui/button (t :file-rn/confirm-proceed) ;; the button is for triple-lowbar only
+                   :class "text-sm p-1 mr-1"
+                   :on-click #(do (reset! *target-format :triple-lowbar)
+                                  (reset! *dir-format (state/get-filename-format repo)) ;; assure it's uptodate
+                                  (write-filename-format! repo :triple-lowbar)
+                                  (reset! *solid-format :triple-lowbar)))])
+     (when (state/developer-mode?)
+       [:div [:hr]
+        (filename-format-select *target-format @*switch-disabled?)
+        (ui/button (t :file-rn/confirm-proceed) ;; the button is for persisting selected format
+                   :disabled (not need-persist?)
+                   :class "text-sm p-1 mr-1"
+                   :on-click #(do (reset! *dir-format (state/get-filename-format repo)) ;; assure it's uptodate
+                                  (write-filename-format! repo @*target-format)
+                                  (reset! *solid-format @*target-format)
+                                  (reset! *switch-disabled? true)))])
+     [:div.cp__settings-files-breaking-changed {:disabled need-persist?} [:hr]
+      (let [rename-items  (->> (vals @*pages)
+                               (map (fn [[page file]]
+                                      (when-let [ret (calc-rename-target page (:file/path file) @*dir-format @*target-format)]
+                                        (merge ret {:page page :file file}))))
+                               (remove nil?))
+            <rename-all   #(async/go (doseq [{:keys [file target status]} rename-items]
+                                       (when (not= status :unreachable)
+                                         (async/<! (p->c (page-handler/rename-file! file target (constantly nil) true)))))
+                                     (<close-modal-on-done))]
+
+        (if (not-empty rename-items)
+          [:div ;; Normal UX stage 2: close stage 1 UI, show the action description as admolition
+           (if (and (= @*solid-format :triple-lowbar)
+                    (= @*dir-format :legacy))
+             (ui/admonition :tip [:p (t :file-rn/need-action)])
+             [:p (t :file-rn/need-action)])
+           [:p
+            (ui/button
```

Comment:
If I click "Update format!", leave the setting and come back, I no longer can see this apply button. This seems like a possible bug as a user may want to think about this and come back to it. Instead when they come back, they see `Well done! No further action required` when they haven't done a rename

Reply:
@logseq-cldwalker Humm, I want to have a clarification on the design:
* Although the format update is optional, we don't hope users "downgrade" the format to the legacy one. 
* The design here hope users can "preview" the in-coming changes required on the format changed. 
* We need to update the config before the files are renamed so the file event can be correctly handled on other clients (although we suggest users to re-index the files on all clients).

> leave the setting and come back, I no longer can see this apply button

Maybe a "Rename Done" button to confirm all renaming operations done would work? But basically we don't hope users to think twice on this - this is a non-revertible operation..

Reply:
I don't think I explained the issue well as there is no downgrade involved. https://www.loom.com/share/e45e561da8b248d48b972f2561799ce7 shows the issue

Reply:
@logseq-cldwalker Yeah, I mean adding an intermediate state `renaming` , and only exit the `renaming` state via clicking the button `Rename Done`.
Alternative solution would be to notice users that they can't return back to the list once they close the modal

Reply:
3-step would be too complex. For now, I've warned the user of the behavior with "These actions are not available once you close this panel". Users can workaround this by removing the config.edn change if they want